server/src/wallet.ts

```diff
@@ -0,0 +1,91 @@
+import { Chain } from '../shovel-config'
+import db from './db'
+
+async function getOwnedBodies(chain: Chain, address?: string) {
+  if (!address) return []
+
+  const problems = await db.query(
+    `
+    WITH latest_transactions AS (
+      SELECT
+          token_id,
+          "to",
+          ROW_NUMBER() OVER (PARTITION BY token_id ORDER BY block_num DESC, tx_idx DESC, log_idx DESC) AS rn
+      FROM
+        problems_transfer
+      WHERE
+        "to" = decode($1, 'hex')
+        AND src_name = '${chain}'
+  ),
+  current_owners AS (
+      SELECT
+          token_id
+      FROM
+          latest_transactions
+      WHERE
+          rn = 1
+  ),
+  latest_body_status AS (
```

Comment:
don't understand what the UNION does here but looks.... nice : )

maybe good to have a little explainer for the queries in case we need to debug them one day?

Reply:
the UNION combines all the rows into one table so you get the latest add/remove event

similar to your `bodyAddRemoveEvents` on the frontend

Reply:
might rename to that